### ISSUE TITLE
Bug: [getter-return] false positives with `Object.defineProperty()` and similar methods

### ISSUE THREAD

**Issue by Grit03** [eslint/eslint#21157]
### Environment

ESLint Playground

### What parser are you using?

Default (Espree)

### What did you do?

```js
// Case 1: `get` in the target object, not in the property descriptor
Object.defineProperty({ get(key) {} }, "name", { value: 1 });
Object.create({ foo: { get(key) {} } }, { bar: { value: 1 } });

// Case 2: `Object` is shadowed, not the global `Object`
function f(Object) {
    Object.defineProperty({}, "k", { get() {} });
}
```

### What did you expect to happen?

No errors should be reported in either case.

Case 1: The get method is in the target object argument, not in the property descriptor argument, so it is not a getter.
Case 2: Object is shadowed by a function parameter, so this is not a call to the global `Object.defineProperty` and the rule should not apply.

Note that the similar rule `no-setter-return` handles both cases correctly with its `isArgumentOfGlobalMethodCall()` and `isPropertyDescriptor()` helper functions: https://github.com/eslint/eslint/blob/d08629382b0a6aaa042823b796e5100e60053b54/lib/rules/no-setter-return.js#L28-L104

### What actually happened?

Errors are reported in all three places:

`2:25 Expected to return a value in method 'get'.`
`3:24 Expected to return a value in method 'get'.`
`7:38 Expected to return a value in method 'get'.`

### Link to Minimal Reproducible Example

https://eslint.org/play/#eyJ0ZXh0IjoiLy8gQ2FzZSAxOiBgZ2V0YCBpbiB0aGUgdGFyZ2V0IG9iamVjdCwgbm90IGluIHRoZSBwcm9wZXJ0eSBkZXNjcmlwdG9yXG5PYmplY3QuZGVmaW5lUHJvcGVydHkoeyBnZXQoa2V5KSB7fSB9LCBcIm5hbWVcIiwgeyB2YWx1ZTogMSB9KTtcbk9iamVjdC5jcmVhdGUoeyBmb286IHsgZ2V0KGtleSkge30gfSB9LCB7IGJhcjogeyB2YWx1ZTogMSB9IH0pO1xuXG4vLyBDYXNlIDI6IGBPYmplY3RgIGlzIHNoYWRvd2VkLCBub3QgdGhlIGdsb2JhbCBgT2JqZWN0YFxuZnVuY3Rpb24gZihPYmplY3QpIHtcbiAgICBPYmplY3QuZGVmaW5lUHJvcGVydHkoe30sIFwia1wiLCB7IGdldCgpIHt9IH0pO1xufVxuIiwib3B0aW9ucyI6eyJydWxlcyI6eyJjb25zdHJ1Y3Rvci1zdXBlciI6WyJlcnJvciJdLCJmb3ItZGlyZWN0aW9uIjpbImVycm9yIl0sImdldHRlci1yZXR1cm4iOlsiZXJyb3IiXSwibm8tYXN5bmMtcHJvbWlzZS1leGVjdXRvciI6WyJlcnJvciJdLCJuby1jYXNlLWRlY2xhcmF0aW9ucyI6WyJlcnJvciJdLCJuby1jbGFzcy1hc3NpZ24iOlsiZXJyb3IiXSwibm8tY29tcGFyZS1uZWctemVybyI6WyJlcnJvciJdLCJuby1jb25kLWFzc2lnbiI6WyJlcnJvciJdLCJuby1jb25zdC1hc3NpZ24iOlsiZXJyb3IiXSwibm8tY29uc3RhbnQtYmluYXJ5LWV4cHJlc3Npb24iOlsiZXJyb3IiXSwibm8tY29uc3RhbnQtY29uZGl0aW9uIjpbImVycm9yIl0sIm5vLWNvbnRyb2wtcmVnZXgiOlsiZXJyb3IiXSwibm8tZGVidWdnZXIiOlsiZXJyb3IiXSwibm8tZGVsZXRlLXZhciI6WyJlcnJvciJdLCJuby1kdXBlLWFyZ3MiOlsiZXJyb3IiXSwibm8tZHVwZS1jbGFzcy1tZW1iZXJzIjpbImVycm9yIl0sIm5vLWR1cGUtZWxzZS1pZiI6WyJlcnJvciJdLCJuby1kdXBlLWtleXMiOlsiZXJyb3IiXSwibm8tZHVwbGljYXRlLWNhc2UiOlsiZXJyb3IiXSwibm8tZW1wdHkiOlsiZXJyb3IiXSwibm8tZW1wdHktY2hhcmFjdGVyLWNsYXNzIjpbImVycm9yIl0sIm5vLWVtcHR5LXBhdHRlcm4iOlsiZXJyb3IiXSwibm8tZW1wdHktc3RhdGljLWJsb2NrIjpbImVycm9yIl0sIm5vLWV4LWFzc2lnbiI6WyJlcnJvciJdLCJuby1leHRyYS1ib29sZWFuLWNhc3QiOlsiZXJyb3IiXSwibm8tZmFsbHRocm91Z2giOlsiZXJyb3IiXSwibm8tZnVuYy1hc3NpZ24iOlsiZXJyb3IiXSwibm8tZ2xvYmFsLWFzc2lnbiI6WyJlcnJvciJdLCJuby1pbXBvcnQtYXNzaWduIjpbImVycm9yIl0sIm5vLWludmFsaWQtcmVnZXhwIjpbImVycm9yIl0sIm5vLWlycmVndWxhci13aGl0ZXNwYWNlIjpbImVycm9yIl0sIm5vLWxvc3Mtb2YtcHJlY2lzaW9uIjpbImVycm9yIl0sIm5vLW1pc2xlYWRpbmctY2hhcmFjdGVyLWNsYXNzIjpbImVycm9yIl0sIm5vLW5ldy1uYXRpdmUtbm9uY29uc3RydWN0b3IiOlsiZXJyb3IiXSwibm8tbm9ub2N0YWwtZGVjaW1hbC1lc2NhcGUiOlsiZXJyb3IiXSwibm8tb2JqLWNhbGxzIjpbImVycm9yIl0sIm5vLW9jdGFsIjpbImVycm9yIl0sIm5vLXByb3RvdHlwZS1idWlsdGlucyI6WyJlcnJvciJdLCJuby1yZWRlY2xhcmUiOlsiZXJyb3IiXSwibm8tcmVnZXgtc3BhY2VzIjpbImVycm9yIl0sIm5vLXNlbGYtYXNzaWduIjpbImVycm9yIl0sIm5vLXNldHRlci1yZXR1cm4iOlsiZXJyb3IiXSwibm8tc2hhZG93LXJlc3RyaWN0ZWQtbmFtZXMiOlsiZXJyb3IiXSwibm8tc3BhcnNlLWFycmF5cyI6WyJlcnJvciJdLCJuby10aGlzLWJlZm9yZS1zdXBlciI6WyJlcnJvciJdLCJuby11bmFzc2lnbmVkLXZhcnMiOlsiZXJyb3IiXSwibm8tdW5kZWYiOlsiZXJyb3IiXSwibm8tdW5leHBlY3RlZC1tdWx0aWxpbmUiOlsiZXJyb3IiXSwibm8tdW5yZWFjaGFibGUiOlsiZXJyb3IiXSwibm8tdW5zYWZlLWZpbmFsbHkiOlsiZXJyb3IiXSwibm8tdW5zYWZlLW5lZ2F0aW9uIjpbImVycm9yIl0sIm5vLXVuc2FmZS1vcHRpb25hbC1jaGFpbmluZyI6WyJlcnJvciJdLCJuby11bnVzZWQtbGFiZWxzIjpbImVycm9yIl0sIm5vLXVudXNlZC1wcml2YXRlLWNsYXNzLW1lbWJlcnMiOlsiZXJyb3IiXSwibm8tdXNlbGVzcy1hc3NpZ25tZW50IjpbImVycm9yIl0sIm5vLXVzZWxlc3MtYmFja3JlZmVyZW5jZSI6WyJlcnJvciJdLCJuby11c2VsZXNzLWNhdGNoIjpbImVycm9yIl0sIm5vLXVzZWxlc3MtZXNjYXBlIjpbImVycm9yIl0sIm5vLXdpdGgiOlsiZXJyb3IiXSwicHJlc2VydmUtY2F1Z2h0LWVycm9yIjpbImVycm9yIl0sInJlcXVpcmUteWllbGQiOlsiZXJyb3IiXSwidXNlLWlzbmFuIjpbImVycm9yIl0sInZhbGlkLXR5cGVvZiI6WyJlcnJvciJdfSwibGFuZ3VhZ2VPcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYUZlYXR1cmVzIjp7fX19fX0=

### Participation

- [x] I am willing to submit a pull request for this issue.

### AI acknowledgment

- [ ] I did not use AI to generate this issue report.
- [x] (If the above is not checked) I have reviewed the AI-generated content before submitting.

### Additional comments

Since both false positives come from the same check in `getter-return`, I think a single PR can fix both — happy to split it if the team prefers.

I'd like to fix them by applying the same validation that `no-setter-return`'s helpers (`isArgumentOfGlobalMethodCall()` / `isPropertyDescriptor()`) already perform: **checking the argument position and verifying that the call references the global object.**

One question about the implementation: `accessor-pairs` also has nearly identical logic,
https://github.com/eslint/eslint/blob/eddbad60158d8cda86830fbe934a9d2282273b32/lib/rules/accessor-pairs.js#L90-L138
so this fix would mean a third copy of the same helpers. Would it be better to move them into `ast-utils.js` for reuse, or should I just duplicate them in `getter-return` for now and leave the extraction for a follow-up?

**Comment by DMartens** [comment]
I agree that three cases are false positives.
I also agree that the helper `isArgumentOfGlobalMethodCall` from `no-setter-return` should be moved to `ast-utils` and be used by the 3 mentioned rules.
In my opinion it would be okay to do this in a single PR as the change is small enough.
Feel free to open a PR for this.
